code/scripting/api/objs/subsystem.cpp

```diff
@@ -20,19 +20,16 @@ bool turret_fire_weapon(int weapon_num, ship_subsys *turret, int parent_objnum,
 namespace scripting {
 namespace api {
 
-ship_subsys_h::ship_subsys_h() : object_h() {
-	ss = NULL;
-}
-ship_subsys_h::ship_subsys_h(object* objp_in, ship_subsys* sub) : object_h(objp_in) {
-	ss = sub;
-}
-bool ship_subsys_h::isSubsystemValid() const { return object_h::IsValid() && objp->type == OBJ_SHIP && ss != nullptr; }
+ship_subsys_h::ship_subsys_h() : objh(), ss(nullptr) {}
+ship_subsys_h::ship_subsys_h(object* objp_in, ship_subsys* sub) : objh(objp_in), ss(sub) {}
+
+bool ship_subsys_h::isValid() const { return objh.isValid() && objh.objp->type == OBJ_SHIP && ss != nullptr; }
 
 void ship_subsys_h::serialize(lua_State* /*L*/, const scripting::ade_table_entry& /*tableEntry*/, const luacpp::LuaValue& value, ubyte* data, int& packet_size) {
 	ship_subsys_h subsys;
 	value.getValue(l_Subsystem.Get(&subsys));
-	const ushort& netsig = subsys.IsValid() ? subsys.objp->net_signature : 0;
-	const int& subsys_index = subsys.isSubsystemValid() ? ship_get_subsys_index(subsys.ss) : -1;
+	const ushort& netsig = subsys.isValid() ? subsys.objh.objp->net_signature : 0;
```

Comment:
This probably needs to check for subsys.objh.isValid(), since it can (and maybe should) still send an invalid subsys of a valid obj.

Reply:
Point.  Changed.